legion/engine/physics/physx/physx_integration_helpers.cpp

```diff
@@ -0,0 +1,85 @@
+#include "physx_integration_helpers.hpp"
+#include <physics/physx/systems/physx_physics_system.hpp>
+#include <physx/PxPhysicsAPI.h>
+
+using namespace physx;
+
+namespace legion::physics
+{
+    physx::PxPhysics* getSDK()
+    {
+        return PhysXPhysicsSystem::getSDK();
+    }
+
+    static inline void toPhysxTransform(physx::PxTransform& pxTransform, const math::vec3& pos, const math::quat& rot)
+    {
+        pxTransform.p = { pos.x,pos.y,pos.z };
+        pxTransform.q = { rot.x,rot.y,rot.z,rot.w };
+    }
+
+    void calculateGlobalAndLocalTransforms(physx::PxTransform& outLocalTransform, physx::PxTransform& outGlobalTransform, const ColliderData& collider, ecs::entity ent)
+    {
+        const math::vec3& pos = *ent.get_component<position>();
+        const math::quat& rot = *ent.get_component<rotation>();
+
+        const math::vec3& localOffset = collider.getOffset();
+        const math::quat& localRot = collider.getRotationOffset();
+
+        toPhysxTransform(outGlobalTransform, pos, rot);
+        toPhysxTransform(outLocalTransform, localOffset, localRot);
```

Comment:
should this not take into account the entity parents? and also should the offset not be added onto the global?

Reply:
yea this should have taken into account parent transforms, wil fix it. As for the offset, it isn't added because the offset is added into the collider when you call PxCreateStatic/Dynamic or when you add the collider itself using add shape